tests/app/tests_app/tests.py

```diff
@@ -33,50 +40,195 @@ def setUpClass(cls):
     def tearDownClass(cls):
         pass
 
+
+class DirectMiddlewaresTests(BaseMiddlewaresTests):
```

Comment:
Since many users will be using channels for websockets, you could use [`WebSocketCommunicator`](https://channels.readthedocs.io/en/stable/topics/testing.html#websocketcommunicator) to create some tests that test the query parameter middleware on websocket connections.

Reply:
Fixed in 9ca4d32